src/scenic/simulators/robosuite/simulator.py

```diff
@@ -0,0 +1,566 @@
+    "sideview": 3,
+    "robot0_robotview": 4,
+    "robot0_eye_in_hand": 5
+}
+
+# Object type mapping
+OBJECT_FACTORIES = {
+    # Primitive objects with customizable size
+    'Ball': lambda cfg: BallObject(
+        name=cfg['name'],
+        size=[cfg.get('radius', 0.02)],
+        rgba=cfg.get('color', [1, 0, 0, 1])
+    ),
+    'Box': lambda cfg: BoxObject(
+        name=cfg['name'],
+        size=cfg.get('size', [0.025, 0.025, 0.025]),
+        rgba=cfg.get('color', [1, 0, 0, 1])
+    ),
+    'Cylinder': lambda cfg: CylinderObject(
+        name=cfg['name'],
+        size=_extract_cylinder_size(cfg),
+        rgba=cfg.get('color', [1, 0, 0, 1])
+    ),
+    'Capsule': lambda cfg: CapsuleObject(
+        name=cfg['name'],
+        size=_extract_cylinder_size(cfg),
+        rgba=cfg.get('color', [1, 0, 0, 1])
+    ),
+    # Complex objects with fixed sizes - no size parameter
+    'Milk': lambda cfg: MilkObject(name=cfg['name']),
+    'Cereal': lambda cfg: CerealObject(name=cfg['name']),
+    'Can': lambda cfg: CanObject(name=cfg['name']),
+    'Bread': lambda cfg: BreadObject(name=cfg['name']),
+    'Hammer': lambda cfg: HammerObject(name=cfg['name']),
+    'SquareNut': lambda cfg: SquareNutObject(name=cfg['name']),
+    'RoundNut': lambda cfg: RoundNutObject(name=cfg['name']),
+    'Bottle': lambda cfg: BottleObject(name=cfg['name'])
+}
+
+
+def _extract_cylinder_size(config: Dict) -> List[float]:
+    """Extract [radius, height] for cylinder-like objects."""
+    if 'radius' in config and 'height' in config:
+        return [config['radius'], config['height']]
+    
+    size = config.get('size', [0.02, 0.02, 0.04])
+    if len(size) == 3:  # Convert from [width, length, height]
+        radius = (size[0] + size[1]) / 4  
+        return [radius, size[2]]
+    elif len(size) == 2:
+        return size
+    return [0.02, 0.04]  # Default
+
+
+class ScenicManipulationEnv(ManipulationEnv):
+    """Scenic-driven manipulation environment."""
+    
+    def __init__(self, scenic_config: Dict, **kwargs):
+        self.scenic_config = scenic_config
+        self.scenic_objects = scenic_config.get('objects', [])
+        self.scenic_tables = scenic_config.get('tables', [])
+        super().__init__(**kwargs)
+    
+    def _load_model(self):
+        """Load models and create arena."""
+        super()._load_model()
+        
+        self._position_robots()
+        mujoco_arena = self._create_arena()
+        mujoco_arena.set_origin([0, 0, 0])
+        
+        self.mujoco_objects = self._create_objects()
+        
+        self.model = ManipulationTask(
+            mujoco_arena=mujoco_arena,
+            mujoco_robots=[robot.robot_model for robot in self.robots],
+            mujoco_objects=self.mujoco_objects
+        )
+    
+    def _position_robots(self):
+        """Position robots based on scenic configuration."""
+        for i, robot_config in enumerate(self.scenic_config.get('robots', [])):
+            if i < len(self.robots):
```

Comment:
I don't see where `self.robots` is initialized (in the superclass `__init__`?), so I'm not sure how this condition can fail. If it does fail, is skipping positioning one of the robots listed in `scenic_config` really the right behavior? I would have thought this would mean there is a problem somewhere.